snapcraft/plugins/go.py

```diff
@@ -144,6 +144,15 @@ def _get_local_go_package(self):
             go_package = os.path.basename(os.path.abspath(self.options.source))
         return go_package
 
+    def _get_local_main_packages(self):
```

Comment:
no, the idea is to read the source code from top to bottom, you start with a high level public method, and following it you define all its helpers. So you can choose if you want a high-level view, or dig into the details. If you first find an internal helper, you don't know the context so you have to jump throw the code finding where it is called in order to understand it.